src/sonarqube-analyzers/sourcemeter-analyzer-csharp/src/main/java/com/sourcemeter/analyzer/csharp/profile/SourceMeterCSharpProfile.java

```diff
@@ -30,33 +30,30 @@
 
 package com.sourcemeter.analyzer.csharp.profile;
 
+import java.io.IOException;
 import java.io.InputStream;
 
-import org.apache.commons.io.IOUtils;
-import org.sonar.api.profiles.RulesProfile;
-import org.sonar.api.profiles.XMLProfileParser;
-import org.sonar.api.utils.ValidationMessages;
-
 import com.sourcemeter.analyzer.base.profile.SourceMeterProfile;
+import com.sourcemeter.analyzer.csharp.core.CSharp;
 
 public class SourceMeterCSharpProfile extends SourceMeterProfile {
 
-    public SourceMeterCSharpProfile(XMLProfileParser parser) {
-        super(parser);
-    }
-
     /**
      * {@inheritDoc}
      */
     @Override
-    public RulesProfile createProfile(ValidationMessages validationMessages) {
-        InputStream input = null;
+    public void define(Context context) {
+        NewBuiltInQualityProfile qualityProfile = context.createBuiltInQualityProfile("SourceMeter Way", CSharp.KEY);
+        InputStream qualityProfileXml = getClass().getResourceAsStream("/SourceMeter_way_default_profile.xml" );
+
+        super.parseXml(qualityProfile, qualityProfileXml);
+
         try {
-            input = getClass().getResourceAsStream(
-                    "/SourceMeter_way_default_profile.xml");
-            return super.createProfile(input, ValidationMessages.create());
-        } finally {
-            IOUtils.closeQuietly(input);
+            qualityProfileXml.close();
```

Comment:
Not just the `close` can throw `IOException`. Why did you remove the `finally` block?

Reply:
Resolved for all five analyzers